### PR TITLE
consensus: exit if wrong password is provided in configuration

### LINKED ISSUE #1312: Print some meaningful error when unable to unlock the configured wallet
I've spent like half an hour debugging the node that didn't want to participate in consensus process just because the node's configuration didn't have any password provided for the wallet. It shouldn't be like that, the node MUST shout out loud something when it can't unlock the configured wallet. And it should do so early in the startup process as a configuration sanity check.

Both for Neo 2 and Neo 3.

### PR DESCRIPTION
**Pull request by fyrchik**
Fix #1312 .

### CODE REVIEW

File: pkg/consensus/consensus.go [hunk 1]
```diff
@@ -299,7 +300,8 @@ func (s *service) getKeyPair(pubs []crypto.PublicKey) (int, crypto.PrivateKey, c
 
 		key, err := keys.NEP2Decrypt(acc.EncryptedWIF, s.Config.Wallet.Password)
 		if err != nil {
-			continue
+			s.log.Fatal("can't unlock account", zap.String("address", address.Uint160ToString(sh)))
+			break
```
Comment (roman-khimov) [inline pkg/consensus/consensus.go:304]: We should fail as early as possible, this one is a configuration error, we should get it like immediately upon node startup. I'd propose moving wallet opening to the network's `NewServer` and checking the password there. The node just won't start at all this way and this is what we want when it's obviously not configured properly.

Of course this more late check also makes sense, but only as an additional one, not the primary.

Comment (roman-khimov) [reply pkg/consensus/consensus.go:304]: Though returning an error from `consensus.NewService` probably also is fine.

Comment (fyrchik) [reply pkg/consensus/consensus.go:322]: Fixed.

File: pkg/consensus/consensus.go [hunk 2]
```diff
@@ -121,6 +122,24 @@ func NewService(cfg Config) (Service, error) {
 		return nil, err
 	}
 
+	// Check that wallet password is correct.
+	vals, err := cfg.Chain.GetValidators()
```
Comment (roman-khimov) [inline pkg/consensus/consensus.go:126]: I'm at 0, I'm not yet elected, still I have a password for some account and it should work.

Comment (fyrchik) [reply pkg/consensus/consensus.go:126]: This is where the check in `getKeyPair` comes into play.

Comment (fyrchik) [reply pkg/consensus/consensus.go:126]: Checking all available accounts is not an option, because password is per-account, not per-wallet.

Comment (fyrchik) [reply pkg/consensus/consensus.go:126]: Well, we can check that password match to at least one account.

File: pkg/consensus/consensus.go [hunk 3]
```diff
@@ -121,6 +122,19 @@ func NewService(cfg Config) (Service, error) {
...
+	var ok bool
+	for _, acc := range srv.wallet.Accounts {
+		err := acc.Decrypt(srv.Config.Wallet.Password)
+		if err == nil {
+			ok = true
+			break
+		}
+	}
+	if !ok {
+		return nil, errors.New("no account with provided password were found")
```
Comment (roman-khimov) [inline pkg/consensus/consensus.go:135]: was

**Comment by roman-khimov** [comment]
And we also need this for master branch.

**Pull request merged**